calyx/frontend/src/ast.rs

```diff
@@ -256,6 +259,37 @@ pub struct StaticGroup {
     pub latency: NonZeroU64,
 }
 
+#[derive(Debug)]
+/// Transitions are either `Unconditional` (will always transition to a future state
+/// no matter the assignments) or `Conditional` (with boolean expressions denoted a
+/// `guard_state_pair` and a default).
+/// Conditions encode a vector of tuples - the guard and the corresponding state to be transitioned to.
+pub enum Transition {
+    Unconditional(u64),
+    Conditional(Vec<(Guard, u64)>),
+}
+
+#[derive(Debug)]
+/// A `FSMRule` is the set of assignments and transitions that corresponds to a state in `fsm`.
+/// Assignments are vectors of wires corresponding to the state index.
+/// Transitions encode the next state logic.
+pub struct FSMRule {
+    pub assignments: Vec<Wire>,
+    pub transition: Transition,
+}
+
+#[derive(Debug)]
+/// A fsm block has a name, attributes, and `FSMRules`
```

Comment:
```suggestion
/// A fsm block has a name, attributes, and `FSMRule`s
```

Pluralizing the Rust identifier.